app/src/main/java/com/android/shelfLife/model/newInvitations/InvitationRepositoryFirestore.kt

```diff
@@ -0,0 +1,117 @@
+package com.android.shelfLife.model.newInvitations
+
+import android.util.Log
+import com.android.shelfLife.model.newhousehold.HouseHold
+import com.android.shelfLife.model.user.User
+import com.google.firebase.Timestamp
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FieldPath
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ListenerRegistration
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.tasks.await
+
+open class InvitationRepositoryFirestore(private val db: FirebaseFirestore) : InvitationRepository {
+
+  internal var listenerRegistration: ListenerRegistration? = null
+  private val _invitations = MutableStateFlow<List<Invitation>>(emptyList())
+  val invitations: StateFlow<List<Invitation>> = _invitations.asStateFlow()
+  private val invitationPath = "invitations"
+
+  /** Removes the real-time listener for invitations. */
+  override fun removeInvitationListener() {
+    listenerRegistration?.remove()
+    listenerRegistration = null
+  }
+
+  /**
+   * Sends an invitation to a user to join a household.
+   *
+   * @param household The household to invite the user to.
+   */
+  override fun sendInvitation(household: HouseHold, invitedUser: User) {
+    val invitedUserId = invitedUser.uid
```

Comment:
maybe it would be good if this mapof was refactored into a function like in the fooditems repository